File: examples/java/advanced/error-handling/test-retry-behavior.java
@Test
void testRetrySuccess() {
    var handler = new RetryHandler();
    var runner = LocalDurableTestRunner.create(Map.class, handler);

    var result = runner.runUntilComplete(Map.of());

    // Should succeed after retries
    assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());
}

File: examples/python/sdk-reference/error-handling/test-retry-behavior.py
@pytest.mark.durable_execution(
    handler=handler_with_retry,
    lambda_function_name="retry_function",
)
def test_retry_success(durable_runner):
    """Test that retries eventually succeed."""
    with durable_runner:
        result = durable_runner.run(input={}, timeout=30)
    
    # Should succeed after retries
    assert result.status is InvocationStatus.SUCCEEDED
